static/src/javascripts/projects/common/modules/commercial/article-body-adverts.js

```diff
@@ -62,19 +72,29 @@ define([
 
             if (config.page.hasInlineMerchandise) {
                 adNames.unshift(['im', 'im']);
-                insertAdAtP(spacefinder.getParaWithSpace(lenientRules));
-            }
-            insertAdAtP(spacefinder.getParaWithSpace(rules));
 
-            if (detect.isBreakpoint({ max: 'tablet' })) {
-                insertAdAtP(spacefinder.getParaWithSpace(rules));
+                inlineMercPromise = spacefinder.getParaWithSpace(lenientRules).then(function (space) {
+                    return insertAdAtP(space);
+                });
+            } else {
+                inlineMercPromise = Promise.resolve(null);
             }
+
+            return inlineMercPromise.then(function () {
```

Comment:
It's absolutely hideous, I know.

`Promise.all` won't work, though, as we want each thing to run synchronously. (i.e., the second promise should only be triggered after the first completed. `all` will run them all in parallel, but give you a promise for when they've all completed.)